cosmos/scheduler/cmd/scheduler.go

```diff
@@ -45,7 +52,23 @@ func (s *Scheduler) Run() {
 	}
 }
 
-func (s *Scheduler) configureIngestionSecret() error {
+func (s *Scheduler) healthCheck() {
+	health := func(w http.ResponseWriter, req *http.Request){
+		ctx := context.Background()
```

Comment:
The context from the request can be reused, this way cancellation (if the connection is severed or the server is shutting down for example) can be cleanly propagated.

Reply:
(or even better a WithTimeout on the request's context, to avoid a hanging mongodb from having health checks take longer than the check interval)

Reply:
Reusing the context makes sense, however If there it is WithTimeout, then can it be reused?

Reply:
Contexts are organised in a tree, so WithTimeout creates a child context of the request context which will get canceled all the same, and also have its own time constraint